aten/src/ATen/native/sparse/SparseFactories.cpp

```diff
@@ -0,0 +1,168 @@
+#include <ATen/Dispatch.h>
+#include <ATen/SparseTensorImpl.h>
+#include <ATen/SparseTensorUtils.h>
+#include <ATen/core/ATen_fwd.h>
+#include <ATen/core/Tensor.h>
+#include <ATen/ops/sparse_coo_tensor.h>
```

Comment:
Guard on `AT_PER_OPERATOR_HEADERS` as in [SparseCsrTensorMath.cpp](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp#L17-L105), which should probably resolve your build error in [no-ops](https://github.com/pytorch/pytorch/runs/6730702999?check_suite_focus=true).